tests/cases/compiler/bigintIndex.ts

```diff
@@ -30,3 +31,11 @@ typedArray[2] = 0xCC;
 const a = {1n: 123};
 const b = {[1n]: 456};
 const c = {[bigNum]: 789};
+{ ({1n: 123}); };
+
+// @filename: c.ts
+const crr = [1, 2, 3] as const;
+
+const { 0n: f } = crr; // bigint should give an index error
+const { 0: d } = crr;
+const { "0": e } = crr;
```

Comment:
For these new tests, I would consider put them into a separate file because they have more to do with parsing/grammar checks than indexing into a value (and I would still add tests around trying to index with `bigint`s into those values). I'd probably put that somewhere under `conformance`.

In that file, consider also adding separate "inner" files, each with a test for an `interface`, a `class`, an `enum`, an an [arbitrary import name](https://github.com/microsoft/TypeScript/pull/49297) like

```ts
import { 0n as foo } from "foo";
```